### DISCUSSION TITLE
Android app for blinko

### DISCUSSION THREAD

**Discussion by pepitoria** [blinkospace/blinko#203] [General]
Hi there, I am an Android developer, and also a selfhoster. I recently discovered blinko and I've been loving it since.

I am thinking in building and android app so I can add stuff to my blinko instance seamlessly in my phone.

Is this something you guys would be interested in? 

As of today My idea is to just be able to add stuff to blinko, and not have a full android client, which is a bit too much for the time I have right now.

**Comment by blinko-space** [comment]
That's great. Any contribution you can make is welcome!The existing api can already do quite a bit of functionality, so if you have any questions about the api, feel free to leave a comment.

**Reply by blinko-space** [reply]
https://blinko.apidocumentation.com/reference#tag/note

**Reply by pepitoria** [reply]
Thanks! Right now I am working with the authorization token in the app, getting it from network calls when using the webapp. But this is a bit hacky, which works for me for now while development.

What I have not seen is a login api method call? how can I get the token providing the username and password so I can effectively login from the app?

I could embed a webview to login and get the token, since we already have an api I wanted to ask.

Thanks!

**Comment by pepitoria** [comment]
Hi there! it took a while, mainly because I have very little time, but I just opened [the repo](https://github.com/pepitoria/BlinkoAndroid) and there is a [v0.1 here](https://github.com/pepitoria/BlinkoAndroid/releases/tag/v0.1)

Download the apk, install it, login with instance url and token.

This is my first time open sourcing something. So any comments, suggestions, or whatever are appreciated!!

Cheers!

**Reply by blinko-space** [reply]
Even if it's very simple, thank you for your contribution. It has been added to the community list of the blinko official documentation. 
[community](https://docs.blinko.space/en/community/community)

**Reply by pepitoria** [reply]
that's awesome, thank you!

**Reply by davidfuckingwebb** [reply]
Hey, I just created an account on github for the first time, after years of using various software, and dealt with their horrible shadow identification captcha which took me like eight tries, just to say thanks!! I heard of Blinko last night, installed it this morning, was wishing there was an Android app this afternoon, and then found this thread. I am remembering how great the opensource community is. Thanks!!

Just installed.

**Reply by pepitoria** [reply]
Hi there! I am glad someone (besides me) is using my app :)
The best way to install it/update it is, for now, using [obtanium](https://github.com/ImranR98/Obtainium). I should probably add that in the Readme file.

If you find any issues or have any comments about the app don't hesitate to say so!

**Reply by tarbib** [reply]
Incredible @pepitoria !! Thanks a lot
Would you consider making it available on F-droid ?

**Reply by pepitoria** [reply]
yeah it was in my non urgent todo list because me, personally, I am using more obtanium than fdroid, but I will try to give it a look soon(ish)! Thanks for the suggestion @tarbib

**Reply by pepitoria** [reply]
Hey @tarbib I have been taking a look to the publish process to F-Droid and from what I read it is not as straight forward or trivial as one would like. While I will work on make it available, could you please open a discussion as [a new Idea in the app's repo](https://github.com/pepitoria/BlinkoAndroid/discussions/new?category=ideas) to keep track of this? I'd really appreciate it!

**Reply by tarbib** [reply]
Done! https://github.com/pepitoria/BlinkoAndroid/discussions/3 :) @pepitoria

**Reply by crisqyxw** [reply]
Awesome app! 
However, my personal view on UI is that rather than trying to replicate the complex and fancy web UI, perhaps following Material You's simple interface like moememos is a more universal choice. (It's a pity that I only know Flutter and can't be of much help 😢)

**Reply by pepitoria** [reply]
Hey! Thank you for the comment! Yes, a redesign is definitely in my todo list although I'm focusing first on functionality with the limited time I have.

I want to finish up a couple of things
- marking todos as done
- improving search (right now you cannot search by two tags at the same time, for Instance)
- improving note editing (specially regarding tags autocomplete)

And then I'll redesign it (unless I get fed up of the current design sooner lol)

Anyway thanks for your comment!

**Comment by wsw70** [comment]
As a side note - is there a "value" (in the sense of functionality, I cannot have a more appropriate word, my intent is absolutely not to downplay the efforts of @pepitoria !) in an app vs the web app? The current version is an installable PWA so it is seamless as well.

The only thing missing is the intent to add it to the "share with ..." panel.

**Reply by pepitoria** [reply]
Probably not! And to be honest I'm glad there is a better alternative since I don't have much time to maintain or improve my app.

Still I'm happy using my own thing 🤣

**Comment by ToolsHere** [comment]
Hi everyone 👋
I came across this discussion while working on a similar Android/Blinko use case. Great work on BlinkoAndroid — it's interesting to see another approach to native Android integration with Blinko.

I've been building BlinkoDigestDroid, which has a slightly different focus: rather than being a full Blinko client, it's mainly designed as a capture/processing tool for Android's Share Sheet.

From YouTube, TikTok, Chrome, X, Instagram or other apps, you can share a URL or text to BlinkoDigestDroid, which extracts useful metadata and sends the resulting Markdown note to a self-hosted Blinko instance through /api/v1/note/upsert.

Some current features:

Android Share Sheet integration
metadata/title/thumbnail extraction
automatic tags
local encrypted storage for Blinko credentials
connection diagnostics

I'm currently looking for feedback from other Blinko users, especially around compatibility with different self-hosted configurations.

Website: https://www.domogeek.ovh/en
APK: https://www.domogeek.ovh/api/download?file=BlinkoDigestDroid.apk

I'd also be interested to compare approaches around authentication and Android sharing if useful.
